lib/src/binary_codec.dart

```diff
@@ -287,6 +298,81 @@ class PostgresBinaryEncoder extends Converter<dynamic, Uint8List?> {
 
     return writer.toBytes();
   }
+
+  /// Encode String / double / int to numeric / decimal  without loosing precision.
+  /// Compare implementation: https://github.com/frohoff/jdk8u-dev-jdk/blob/da0da73ab82ed714dc5be94acd2f0d00fbdfe2e9/src/share/classes/java/math/BigDecimal.java#L409
+  Uint8List _encodeNumeric(String value) {
+    value = value.trim();
+    var signByte = 0x0000;
+    if (value.toLowerCase() == 'nan') {
+      signByte = 0xc000;
+      value = '';
+    } else if (value.startsWith('-')) {
+      value = value.substring(1);
+      signByte = 0x4000;
+    } else if (value.startsWith('+')) {
+      value = value.substring(1);
+    }
+    final numericPattern = RegExp(r'^(\d*)(\.\d*)?$');
```

Comment:
Similarly, let's move the `RegExp` expressions to a top-level final field, e.g. `_numericPatternRegExp`, `_patternLeadingZerosRegExp` and so on... That way it will be initialized and parsed only once.